### PR TITLE
Ebb fixes for fpic and ppc32 handler

### PR DESCRIPTION
**Pull request by pmur**
This started as an attempt to fix the -fpic issue. I uncovered several issues with the ppc32 ebb handlers when I noticed the backtrace test was failing under ppc32. This should address issues #10 and #11.

This should resolve them. The PPC64 handler will likely need some similar fixes (Save/restore VSCR, and saving 512B stack in the VR handler).

### CODE REVIEW

**Comment by pmur** [comment]
More updates. I've greatly overhauled the clobber test, and simplified the PPC32 vector load/stores based on the ABI, and fixed a handful of bugs in the PPC64 handler.

**Comment by pmur** [comment]
Another update. I removed redundant ASM directives in the updated test as only P8 supports EBB. I've also unified the ppc32/ppc64 handlers into one file. It simplifies things greatly as they are verify similar.

File: ebb/ebb-callback-ppc32.S [hunk 1]
```diff
@@ -36,135 +36,202 @@
...
- *   r1 + STACK_OFFSET+152:408   |  FRP F1:F31
- *   r1 + STACK_OFFSET+148       |  FPSCR
  *   --------------  __paf_ebb_callback_handler_gpr_[tcb|tls]  -----------------
- *   r1 + STACK_OFFSET+16:144    |  GRP r0,r3-r31
- *   r1 + STACK_OFFSET+12        |  XER
- *   r1 + STACK_OFFSET+8         |  CTR
- *   r1 + STACK_OFFSET+4         |  LR
- *   r1 + STACK_OFFSET           |  CR
+ *   r1 + 152:160   |  Alignment padding
+ *   r1 + 32:152    |  GRP r0,r3-r31
```
Comment (gftg85) [inline ebb/ebb-callback-ppc32.S]: Is this a typo? GPR instead of GRP?

File: ebb/ebb-callback-ppc32.S [hunk 2]
```diff
@@ -36,135 +36,202 @@
...
+ *   r1 + 28        |  XER
+ *   r1 + 24        |  CTR
+ *   r1 + 20        |  LR
+ *   r1 + 16        |  CR
+ *
+ *   --------------  __paf_ebb_callback_handler_fpr_[tcb|tls]  -----------------
+ *   r1 + 408       |  FPSCR (64b store)
+ *   r1 + 152:408   |  FRP f0:f31
```
Comment (gftg85) [inline ebb/ebb-callback-ppc32.S]: FPR instead of FRP?

File: ebb/ebb-callback.S [hunk 3]
```diff
@@ -0,0 +1,613 @@
+/* Event-Based Branch Facility API.  PowerPC32 callback handler.
+ *
+ * Copyright IBM Corp. 2013,2015
```
Comment (tuliom) [inline ebb/ebb-callback.S]: I guess this should be 2013-2015.

File: ebb/ebb-callback.S [hunk 4]
```diff
@@ -0,0 +1,613 @@
+/* Event-Based Branch Facility API.  PowerPC32 callback handler.
```
Comment (tuliom) [inline ebb/ebb-callback.S]: Is this really PowerPC32 only?

File: ebb/tests/test_ebb_backtrace.c [hunk 5]
```diff
@@ -1,6 +1,6 @@
 /* Event-Based Branch Facility Tests.
  *
- * Copyright IBM Corp. 2013
+ * Copyright IBM Corp. 2013,2015
```
Comment (tuliom) [inline ebb/tests/test_ebb_backtrace.c]: Same thing: 2013-2015

File: ebb/tests/test_ebb_pmu_clobber_hooks.S [hunk 6]
```diff
@@ -0,0 +1,514 @@
...
+	CHECK_FPSCR(vsr)
+
+	li r3,0
+.Ldone_vsr:
+	REST_GP_SPRS REST_FPSCR REST_VRSAVE REST_VSCR
+	REST_VSRS REST_GPRS
+	EPILOGUE(vsr)
+END(ebb_test_vsr)
+
+#if 0
```
Comment (tuliom) [inline ebb/tests/test_ebb_pmu_clobber_hooks.S]: What is this doing here?

File: ebb/ebb-callback.S (hunk 4 above)
Comment (pmur) [inline ebb/ebb-callback.S]: It covers all three ABI's now. That comment should be updated. I've tested and verified the generated code on all three.

**Comment by pmur** [comment]
I figured out I wasn't properly setting up the TOC for elfv2 and implemented the clobber functions which had been previously commented out. They now clobber all the common ABI volatile registers.

Otherwise, I addressed the minor comment comments.

File: ebb/tests/test_ebb_backtrace.c (hunk 5 above)
Comment (tuliom) [inline ebb/tests/test_ebb_backtrace.c]: This copyright entry still needs to be fixed.

**Comment by tuliom** [comment]
With that small copyright fix, this will be good to me.

**Comment by pmur** [comment]
Rebased onto master with the requested copyright formatting fix to ebb/test_ebb_backtrace.c.

Thanks all for the review!

**Pull request closed without merging**